src/main/java/school/hei/haapi/model/validator/UpdateFeeValidator.java

```diff
@@ -0,0 +1,68 @@
+package school.hei.haapi.model.validator;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Component;
+import school.hei.haapi.model.Fee;
+import school.hei.haapi.model.exception.BadRequestException;
+import school.hei.haapi.repository.FeeRepository;
+
+@Component
+@AllArgsConstructor
+public class UpdateFeeValidator implements Consumer<Fee> {
+  public final FeeRepository feeRepository;
+
+  public void accept(List<Fee> fees) {
+    fees.forEach(this::accept);
+  }
+
+  @Override
+  public void accept(Fee fee) {
+    Set<String> violationMessages = new HashSet<>();
+    if (fee.getStudent() == null) {
+      violationMessages.add("Student is mandatory");
+    }
+    if (fee.getDueDatetime() == null) {
+      violationMessages.add("Due datetime is mandatory");
+    }
+    if (fee.getTotalAmount() < 0) {
+      violationMessages.add("Total amount must be positive");
+    }
+    if (fee.getId() == null) {
+      violationMessages.add("Id is mandatory");
+    } else {
+      Fee originalFee = feeRepository.findById(fee.getId()).orElse(null);
+      if (originalFee == null) {
```

Comment:
use Optional here.

equivalent of `originalFee == null` is `originalFee.isEmpty()`